### PR TITLE
refact(vg_lite): abstract common pending release logic

### PR DESCRIPTION
**Pull request by FASTSHIFT**
### Description of the feature or fix

Generalized delayed release logic.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.10](https://github.com/szepeviktor/astyle/releases/tag/v3.4.10) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/draw/vg_lite/lv_vg_lite_pending.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ **********************/
+
+typedef struct _lv_vg_lite_pending_t lv_vg_lite_pending_t;
+
+typedef void (*lv_vg_lite_pending_free_cb_t)(void * obj, void * user_data);
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
```
Comment (kisvegabor) [inline src/draw/vg_lite/lv_vg_lite_pending.h:36]: Please add comments here

**Review by kisvegabor — COMMENTED** [review]
I see that you add/remove items to the pending arrays, but can't see where you actually use them, i.e. where you do something with the saved items.

**Comment by FASTSHIFT** [comment]
> I see that you add/remove items to the pending arrays, but can't see where you actually use them, i.e. where you do something with the saved items.

It's here: 
https://github.com/lvgl/lvgl/pull/5756/files#diff-a7fd0e4901fa5b7f6cff798f9ae991251b63efe81d4f1a23d6a1911dac4b04a7R175
https://github.com/lvgl/lvgl/pull/5756/files#diff-8bcb9672aff4be3e779cac54c1c52e2bd88dceb6dddc7d7a976c6235db470d12R202

File: src/draw/vg_lite/lv_vg_lite_pending.h (hunk 1 above)
Comment (FASTSHIFT) [reply src/draw/vg_lite/lv_vg_lite_pending.h:36]: Done.

**Comment by kisvegabor** [comment]
What I can't see is why is it good to add something to the pending list. When/Where do you read the list, and do something with it?

**Comment by FASTSHIFT** [comment]
> What I can't see is why is it good to add something to the pending list. When/Where do you read the list, and do something with it?

Since the cmd buffer of vg-lite only records the address of the image, after submitting the image to the GPU for rendering, we need to prevent the image being rendered from being automatically released by the cache.
Therefore, we need to wait until the GPU is actually rendered before marking the image as releasable. Since rendering is asynchronous, a pending list is needed to record the image information to be released, and the image references are released uniformly in `lv_vg_lite_finish`.

**Review by kisvegabor — APPROVED** [review]

**Comment by kisvegabor** [comment]
Clear now, thanks!

**Review by XuNeo — APPROVED** [review]

**Pull request merged**